src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2444,6 +2453,26 @@ void ApplicationManagerImpl::HeadUnitReset(
   }
 }
 
+void ApplicationManagerImpl::ClearAppsPersistentData() {
+  const std::string apps_info_storage_file = get_settings().app_info_storage();
+  file_system::DeleteFile(apps_info_storage_file);
+
+  const ApplicationSet& accessor = applications().GetData();
+
+  ApplicationSetConstIt it_app_list = accessor.begin();
```

Comment:
@dev-gh The logic is a bit different as in MASTER_RESET SDL fully cleans storage_folder as well as app_icons folder and app_info storage.